database.cc

```diff
@@ -717,6 +717,16 @@ future<> database::update_keyspace(const sstring& name) {
         auto tmp_ksm = db::schema_tables::create_keyspace_from_schema_partition(v);
         auto new_ksm = ::make_lw_shared<keyspace_metadata>(tmp_ksm->name(), tmp_ksm->strategy_name(), tmp_ksm->strategy_options(), tmp_ksm->durable_writes(),
                         boost::copy_range<std::vector<schema_ptr>>(ks.metadata()->cf_meta_data() | boost::adaptors::map_values), std::move(ks.metadata()->user_types()));
+
+        bool old_durable_writes = ks.metadata()->durable_writes();
+        bool new_durable_writes = new_ksm->durable_writes();
+        if (old_durable_writes != new_durable_writes) {
+            for (auto& [cf_name, cf_schema] : new_ksm->cf_meta_data()) {
+                auto& cf = find_column_family(cf_schema);
+                cf.set_durable_writes(new_durable_writes);
+            }
```

Comment:
Note that since this function already returns a future, you could have yielded here (perhaps more easily after converting to a coroutine). Still, I don't think it's necessary here.